test/testNestingDepth.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+from .testHelpers import get_cpp_function_list
+
+
+class TestCppNestingDepth(unittest.TestCase):
+
+    def test_one_function_with_no_nd_condition(self):
+        result = get_cpp_function_list("int fun(){}")
+        self.assertEqual(0, result[0].max_nesting_depth)
+
+    def test_one_function_with_one_nd_condition(self):
+        result = get_cpp_function_list("int fun(){if(a){xx;}}")
+        self.assertEqual(1, result[0].max_nesting_depth)
+
+    def test_one_function_nd_with_question_mark(self):
+        result = get_cpp_function_list("int fun(){return (a)?b:c;}")
+        self.assertEqual(1, result[0].max_nesting_depth)
+
+    def test_one_function_nd_with_forever_loop(self):
```

Comment:
I'm thinking what he's saying is true but CCN and Nesting Depth metrics never mentioned infinite loop. And your example reveals the problem in our algorithm for these metric.
We should fix this issue and modify our code. We can ignore infinite loop statement where they have hard coded value of `true` in them. But we still can not identify some of them in some cases, not just `for(;;){}`. 

For Example:
`   a=1
    if (a==1){
        do_something()
    }`
Can you think of a solution other than preventing some of those and catch specific cases we can identify in every language?